CedarJavaFFI/src/interface.rs

```diff
@@ -613,19 +626,43 @@ pub fn parseEntityUID<'a>(mut env: JNIEnv<'a>, _: JClass, obj: JString<'a>) -> j
     };
     r
 }
+pub fn entity_uid_str(
+    euid_str: &str,
+) -> Result<HashMap<String, String>> {
+    let cedar_euid = EntityUid::from_str(euid_str)?;
+    let mut result = HashMap::new();
+    result.insert("id".to_string(), format!("{:?}", cedar_euid.type_id()));
+    result.insert("type".to_string(), cedar_euid.type_name().to_string());
+    Ok(result)
+}
 
 fn parse_entity_uid_internal<'a>(
```

Comment:
We are changing FFI's contract with CedarJava here. Currently, the corresponding CedarJava method `private static native Optional<EntityUID> parseEntityUID(String src)` is expecting a return of type `Optional<EntityUID`. Here, we are updating it to `Map<String,String>`. But because we haven't updated the method in CedarJava, calling this method will fail.